### PR TITLE
[REVIEW] mdspan-ify make_regression

### PR DESCRIPTION
**Pull request by mhoemmen**
Add an overload of `make_regression` that takes mdspan, instead of raw pointers.  The overload does not increase generality
(e.g., it still requires row-major mdspan).

Part of #535.

### CODE REVIEW

File: cpp/test/random/make_regression.cu [hunk 1]
```diff
@@ -161,4 +163,124 @@ TEST_P(MakeRegressionTestD, Result)
...
+                    coef_mat,
+                    params.bias,
+                    params.effective_rank,
+                    tail_strength,
+                    noise,
+                    params.shuffle,
+                    params.seed,
+                    params.gtype);
+
+    // FIXME (mfh 2022/09/07) This test passes even if I don't call make_regression.
```
Comment (cjnolet) [inline cpp/test/random/make_regression.cu:203]: Can we make a github issue for this and reference it here just to keep it on the radar?

**Review by cjnolet — CHANGES_REQUESTED** [review]
Thanks for starting on the `raft::random` namespace updates so quickly! Just a couple very minor things.

**Comment by cjnolet** [comment]
rerun tests

File: cpp/test/random/make_regression.cu (hunk 1 above)
Comment (mhoemmen) [reply cpp/test/random/make_regression.cu]: @cjnolet Done! https://github.com/rapidsai/raft/issues/814

**Review by cjnolet — APPROVED** [review]
LGTM. Thanks @mhoemmen!

File: cpp/include/raft/random/make_regression.cuh [hunk 2]
```diff
@@ -100,6 +102,82 @@ void make_regression(const raft::handle_t& handle,
...
+ * @param[in]   noise           Standard deviation of the Gaussian noise
+ *                              applied to the output
+ * @param[in]   shuffle         Shuffle the samples and the features
+ * @param[in]   seed            Seed for the random number generator
+ * @param[in]   type            Random generator type
+ */
+template <typename DataT, typename IdxT>
+void make_regression(
+  const raft::handle_t& handle,
+  raft::device_matrix_view<DataT, raft::matrix_extent<IdxT>, raft::row_major> out,
```
Comment (divyegala) [inline cpp/include/raft/random/make_regression.cuh:138]:
```suggestion
  raft::device_matrix_view<DataT, IdxT, raft::row_major> out,
```

Comment (mhoemmen) [reply cpp/include/raft/random/make_regression.cuh:138]: @divyegala Thanks!  I wonder why that even compiled....

**Comment by cjnolet** [comment]
@gpucibot merge

**Comment by mhoemmen** [comment]
rerun tests

**Comment by mhoemmen** [comment]
rerun tests

**Comment by cjnolet** [comment]
@gpucibot merge

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**